x-pack/plugins/security_solution/public/detections/pages/detection_engine/rules/all/bulk_actions/use_bulk_actions.tsx

```diff
@@ -193,13 +200,33 @@ export const useBulkActions = ({
         closePopover();
         startTransaction({ name: BULK_RULE_ACTIONS.EXPORT });
 
-        await executeRulesBulkAction({
+        const response = await executeRulesBulkAction({
           visibleRuleIds: selectedRuleIds,
           action: BulkAction.export,
           setLoadingRules,
           toasts,
           search: isAllSelected ? { query: filterQuery } : { ids: selectedRuleIds },
         });
+
+        // if response null, likely network error happened and export rules haven't been received
+        if (!response) {
+          return;
+        }
+
+        const details = await getExportedRulesDetails(response);
+
+        // if there are failed exported rules, show modal window to users.
+        // they can either cancel action or proceed with export of succeeded rules
+        if (
+          (await showBulkActionConfirmation(
+            transformExportDetailsToDryRunResult(details),
+            BulkAction.export
+          )) === false
+        ) {
+          return;
+        }
```

Comment:
Nit: maybe a subjective opinion, but I don't find such constructs very readable and would always extract an `await ...` call into its own statement.

Reply:
fine by me, but if it helps, I moved it out from condition to make it more readable.